core/src/ModelArray.cpp

```diff
@@ -37,8 +37,12 @@ ModelArray::ModelArray(const ModelArray& orig)
 
 ModelArray& ModelArray::operator=(const ModelArray& orig)
 {
-    type = orig.type;
-    setData(orig.m_data);
+    if (orig.nComponents() == 1 && nComponents() != 1) {
+        component(0) = orig.data();
```

Comment:
```suggestion
        component(0) = orig.data();
        for ( int n=1; n< nComponents(); n++) {
            component(n).setZero();
```

Like this, assigning the mean component resets the higher-order components. The idea is to avoid ending up with uninitialised values when assigning.

Reply:
The use case for this operator is copying back the updated DG0 values after the column physics has run. If the other components were set to 0, as in the suggested change, then the higher DG components would be set to 0. And there is no other copy of the array that the dynamics uses any more.

Reply:
Yes, that's a good point. I was only thinking about what happens when the initial conditions are set. But in this case, we can probably trust that the ``resize`` operation has set everything to zero (as it currently does).

So, ignore this suggestion.